python/lsst/meas/algorithms/treecorrUtils.py

```diff
@@ -0,0 +1,228 @@
+        ),
+        default=4,
+        optional=True,
+        check=lambda x: x > 0,
+    )
+
+    metric = ChoiceField[str](
+        doc=(
+            "Which metric to use for distance measurements. For details, see "
+            "https://rmjarvis.github.io/TreeCorr/_build/html/metric.html"
+        ),
+        default="Euclidean",
+        optional=True,
+        allowed={
+            "Euclidean": "straight-line Euclidean distance between two points",
+            "FisherRperp": (
+                "the perpendicular component of the distance, "
+                "following the definitions in "
+                "Fisher et al, 1994 (MNRAS, 267, 927)"
+            ),
+            "OldRperp": (
+                "the perpendicular component of the distance using the "
+                "definition of Rperp from TreeCorr v3.x."
+            ),
+            "Rlens": (
+                "Distance from the first object (taken to be a lens) to "
+                "the line connecting Earth and the second object "
+                "(taken to be a lensed source)."
+            ),
+            "Arc": "the true great circle distance for spherical coordinates.",
+            "Periodic": "Like ``Euclidean``, but with periodic boundaries.",
+        },
+    )
+
+    bin_type = ChoiceField[str](
+        doc="What type of binning should be used?",
+        default="Log",
+        optional=True,
+        allowed={
+            "Log": (
+                "Logarithmic binning in the distance. The bin steps will "
+                "be uniform in log(r) from log(min_sep) .. log(max_sep)."
+            ),
+            "Linear": (
+                "Linear binning in the distance. The bin steps will be "
+                "uniform in r from min_sep .. max_sep."
+            ),
+            "TwoD": (
+                "2-dimensional binning from x = (-max_sep .. max_sep) "
+                "and y = (-max_sep .. max_sep). The bin steps will be "
+                "uniform in both x and y. (i.e. linear in x,y)"
+            ),
+        },
+    )
+
+    var_method = ChoiceField[str](
+        doc="Which method to use for estimating the variance",
+        default="shot",
+        optional=True,
+        allowed={
+            method: method
+            for method in [
+                "shot",
+                "jackknife",
+                "sample",
+                "bootstrap",
+                "marked_bootstrap",
+            ]
+        },
+    )
+
+    npatch = Field[int](
+        doc="How many patches to split the catalog into for the purpose of "
+        "jackknife variance or other options that involve running via "
+        "patches (boostrap, marked_boostrap etc.)",
+        default=1,
+        optional=True,
+    )
+
+    num_bootstrap = Field[int](
+        doc=(
+            "How many bootstrap samples to use for the 'bootstrap' and 'marked_bootstrap' var methods."
+        ),
+        default=500,
+        optional=True,
+    )
+
+    rng = Field[int](
```

Comment:
Post merge comment: This should remain as `rng_seed`. `rng` typically refers to a `np.RandomState` instance, and that's certainly what `treecorr` means. `rng` itself is not nicely persistable and settable as a config file. `rng_seed` makes it explicit that it is the integer value used to seed that instance.